printing/printer.ml

```diff
@@ -741,7 +739,8 @@ let pr_subgoals ?(pr_first=true) ?(diffs=false) ?os_map
       pr_rec 1 (g::l)
   in
   let pr_evar_info gl sigma seeds =
-    print_evar_constraints gl sigma ++ print_dependent_evars gl sigma seeds
+    let first_goal = if pr_first then gl else None in
+    print_evar_constraints gl sigma ++ print_dependent_evars first_goal sigma seeds
```

Comment:
Of course not. The `None` is inside an `if`, therefore `gl` must have (and always had) the type `Goal.goal option`. Otherwise the compiler would complain about incompatible types in this line.

Reply:
You can see the typing error, if you annotate the `gl` argument of `print_dependent_evars` in `printer.ml` with the type that you require in `printer.mli` in 912527f816.